### PR TITLE
Extract environments frontend from the main SDK

### PR DESCRIPTION
**Pull request by cloderic**
relates to #159 

The general objective is to move from having the environment frontend defined as part of the "main" web app to having them defined as "plugin" coming with the python side of the environment. 

Once this is achieved, it means we can fully decouple the environments from the SDK and then publish the SDK as a library that can be used independently for multiple environments, we can also imagine publishing cogment verse environments as libraries that can be installed when needed. Finally this ground work opens the door to having full web UI instead of an image based render.

---

Base documentation is done, what's missing will be reference documentation of the `@cogment/cogment-verse` SDK to implement a front end and a tutorial for environment implementation.

### CODE REVIEW

File: environments/gym/web/dist/GymCartPole.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import{useCallback as m,useState as A}from"react";import{useDocumentKeypressListener as N,usePressedKeys as R,PlayObserver as S,useRealTimeUpdate as y,TEACHER_ACTOR_CLASS as O,OBSERVER_ACTOR_CLASS as h,Button as E,FpsCounter as T,KeyboardControlList as _,serializePlayerAction as r,TEACHER_NOOP_ACTION as g,Space as v,SimplePlay as w}from"@cogment/cogment-verse";import{jsx as s,jsxs as f}from"react/jsx-runtime";var o=new v({discrete:{n:2}}),L=({sendAction:e,fps:t=30,actorParams:l,...p})=>{let u=l?.className,[n,c]=A(!0),i=m(()=>c(C=>!C),[c]);N("p",i);let a=R(),P=m(C=>{if(a.size===0&&u===O){e(g);return}if(a.has("ArrowLeft")){e(r(o,0));return}else if(a.has("ArrowRight")){e(r(o,1));return}e(r(o,0))},[a,e,u]),{currentFps:d}=y(P,t,n);return f("div",{...p,children:[f("div",{className:"flex flex-row gap-1",children:[s(E,{className:"flex-1",onClick:i,children:n?"Resume":"Pause"}),s(T,{className:"flex-none w-fit",value:d})]}),s(_,{items:[["Left/Right Arrows","push cart to the left/right"],["p","Pause/Unpause"]]})]})},x=({actorParams:e,...t})=>e?.className===h?s(S,{actorParams:e,...t}):s(w,{actorParams:e,...t,controls:L}),k=x;export{k as default};
```
Comment (wduguay-air) [inline environments/gym/web/dist/GymCartPole.js:1]: For gym web components, why do we have all files as Gym[...] and without Gym?

File: environments/petting_zoo/web/dist/chunk-ON5OQYWL.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+var g=Object.create;var f=Object.defineProperty;var h=Object.getOwnPropertyDescriptor;var i=Object.getOwnPropertyNames;var j=Object.getPrototypeOf,k=Object.prototype.hasOwnProperty;var m=(b,a)=>()=>(a||b((a={exports:{}}).exports,a),a.exports);var l=(b,a,c,e)=>{if(a&&typeof a=="object"||typeof a=="function")for(let d of i(a))!k.call(b,d)&&d!==c&&f(b,d,{get:()=>a[d],enumerable:!(e=h(a,d))||e.enumerable});return b};var n=(b,a,c)=>(c=b!=null?g(j(b)):{},l(a||!b||!b.__esModule?f(c,"default",{value:b,enumerable:!0}):c,b));export{m as a,n as b};
```
Comment (wduguay-air) [inline environments/petting_zoo/web/dist/chunk-ON5OQYWL.js:1]: Is this chunk file to be removed? I don't see any in the other environment web components.

**Review by wduguay-air — COMMENTED** [review]
Looks very good overall. Some questions on web components dist files. 

Also, I see that you moved documentation for specific environments under docs/develop. Could you also move the overcooked_ai doc to the same location. It is under docs/integration/overcooked_ai.md.

File: environments/gym/web/dist/GymCartPole.js (hunk 1 above)
Comment (cloderic) [reply environments/gym/web/dist/GymCartPole.js:1]: Hum that's because I forgot to clean it up :)

Comment (cloderic) [reply environments/gym/web/dist/GymCartPole.js]: fixed

File: environments/petting_zoo/web/dist/chunk-ON5OQYWL.js (hunk 2 above)
Comment (cloderic) [reply environments/petting_zoo/web/dist/chunk-ON5OQYWL.js:1]: no it's good here, `esbuild` can splits significant shared part between scripts in chunks. It was only able to do it for PettingZoo

**Comment by cloderic** [comment]
> Also, I see that you moved documentation for specific environments under docs/develop. Could you also move the overcooked_ai doc to the same location. It is under docs/integration/overcooked_ai.md.

ah ok it wasn't linked anywhere I completely overlooked it, it's fixed

**Review by wduguay-air — APPROVED** [review]

**Pull request merged**